pkg/api/v1beta3/types.go

```diff
@@ -1066,3 +1066,60 @@ type List struct {
 
 	Items []runtime.RawExtension `json:"items" description:"list of objects"`
 }
+
+// The following identify resource constants for Kubernetes object types
+const (
+	// Pods, number
+	ResourcePods ResourceName = "pods"
+	// Services, number
+	ResourceServices ResourceName = "services"
+	// ReplicationControllers, number
+	ResourceReplicationControllers ResourceName = "replicationcontrollers"
+	// ResourceQuotas, number
+	ResourceQuotas ResourceName = "resourcequotas"
+)
+
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+	// Hard is the set of desired hard limits for each named resource
+	Hard ResourceList `json:"hard,omitempty"`
+}
+
+// ResourceQuotaStatus defines the enforced hard limits and observed use
+type ResourceQuotaStatus struct {
+	// Hard is the set of enforced hard limits for each named resource
+	Hard ResourceList `json:"hard,omitempty"`
+	// Used is the current observed total usage of the resource in the namespace
+	Used ResourceList `json:"used,omitempty"`
+}
+
+// ResourceQuota sets aggregate quota restrictions enforced per namespace
+type ResourceQuota struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired quota
+	Spec ResourceQuotaSpec `json:"spec,omitempty"`
+
+	// Status defines the actual enforced quota and its current usage
+	Status ResourceQuotaStatus `json:"status,omitempty"`
+}
+
+// ResourceQuotaUsage captures system observed quota status per namespace
+// It is used to enforce atomic updates of a backing ResourceQuota.Status field in storage
+type ResourceQuotaUsage struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Status defines the actual enforced quota and its current usage
+	Status ResourceQuotaStatus `json:"status,omitempty"`
+}
```

Comment:
I treated it like a binding.  The actor that tracks usage is different than the actor that sets the hard quota limits.  

Sent from my iPhone

> On Jan 26, 2015, at 7:46 PM, Eric Tune <notifications@github.com> wrote:
> 
> In pkg/api/v1beta3/types.go:
> 
> > +	// Spec defines the desired quota
> > +	Spec ResourceQuotaSpec `json:"spec,omitempty"`
> > +
> > +	// Status defines the actual enforced quota and its current usage
> > +	Status ResourceQuotaStatus `json:"status,omitempty"`
> > +}
> > +
> > +// ResourceQuotaUsage captures system observed quota status per namespace
> > +// It is used to enforce atomic updates of a backing ResourceQuota.Status field in storage
> > +type ResourceQuotaUsage struct {
> > +	TypeMeta   `json:",inline"`
> > +	ObjectMeta `json:"metadata,omitempty"`
> > +
> > +	// Status defines the actual enforced quota and its current usage
> > +	Status ResourceQuotaStatus `json:"status,omitempty"`
> > +}
> Note to self: come back and figure out why ResourceQuotaUsage is own object, versus just a ResourceQuota with Spec unset by convention.
> 
> —
> Reply to this email directly or view it on GitHub.
>